tests/core/io/test_resource_uid.h

```diff
@@ -47,6 +47,27 @@ TEST_CASE("[ResourceUID] Must encode/decode maximum/minimum UID correctly") {
 	CHECK_MESSAGE(ResourceUID::get_singleton()->text_to_id("uid://a") == 0, "Minimum UID must decode correctly.");
 }
 
+TEST_CASE("[ResourceUID] Must encode and decode invalid UIDs correctly") {
+	ResourceUID *rid = ResourceUID::get_singleton();
+	CHECK_MESSAGE(rid->id_to_text(-1) == "uid://<invalid>", "Invalid UID -1 must encode correctly.");
+	CHECK_MESSAGE(rid->text_to_id("uid://<invalid>") == -1, "Invalid UID -1 must decode correctly.");
```

Comment:
You could also add a test for string that doesn't include `uid://`. They are invalid too.

Reply:
Added one